dojo/tools/rusty_hog/parser.py

```diff
@@ -0,0 +1,89 @@
+import json
+
+from dojo.models import Finding
+
+
+class RustyhogParser(object):
+
+    def get_scan_types(self):
+        return ["Rusty Hog Scan"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return scan_type  # no custom label for now
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Rusty Hog Scan - JSON Report"
+
+    def get_findings(self, json_output, test):
+        tree = json.load(json_output)
+        return self.get_items(tree, test)
+
+    def parse_json(self, json_output):
+        try:
+            data = json_output.read()
+            try:
+                tree = json.loads(str(data, 'utf-8'))
+            except:
+                tree = json.loads(data)
+        except:
+            raise Exception("Invalid format")
+
+        return tree
+
+    def get_items(self, tree, test):
+        items = {}
+        for node in tree:
+            item = get_item(node, test)
+            unique_key = "Finding {}".format(tree.index(node))
+            items[unique_key] = item
+
+        return list(items.values())
+
+
+def get_item(vulnerability, test):
```

Comment:
Can you find out if data is coming from a subparser that's not supported and throw an exception?